application-tour-test/application-tour-test-tests/src/test/it/org/xwiki/contrib/tour/test/ui/TourApplicationTest.java

```diff
@@ -196,39 +196,41 @@ private void tourBoundToPage() throws Exception
         assertFalse(secondPage.isTourDisplayed());
         assertTrue(secondPage.hasResumeButton());
 
-        // Resume (to step 4)
+        // Resume (to step 1, as we have ended the tour)
         secondPage.resume();
         assertTrue(secondPage.isTourDisplayed());
         assertFalse(secondPage.hasResumeButton());
-        assertEquals("Title 4", secondPage.getStepTitle());
-        assertEquals("Step 4", secondPage.getStepDescription());
-        assertFalse(secondPage.hasNextStep());
+        assertEquals("Tour", secondPage.getStepTitle());
+        assertEquals("Description", secondPage.getStepDescription());
+        assertTrue(secondPage.hasNextStep());
+        assertFalse(secondPage.hasPreviousStep());
+        assertFalse(secondPage.hasEndButton());
+
+        // Go one step forward
+        secondPage.nextStep();
+        assertEquals("Title 2", secondPage.getStepTitle());
+        assertEquals("Step 2", secondPage.getStepDescription());
+        assertTrue(secondPage.hasNextStep());
         assertTrue(secondPage.hasPreviousStep());
-        assertTrue(secondPage.hasEndButton());
+        assertFalse(secondPage.hasEndButton());
 
-        // Close
+        // Close the tour (it should start from where we left it)
         secondPage.close();
         assertFalse(secondPage.isTourDisplayed());
         assertTrue(secondPage.hasResumeButton());
 
-        // Go to an other page and then go back
-        TourHomePage.gotoPage();
-        secondPage = PageWithTour.gotoPage("TourCode", "TourClass");
-        assertFalse(secondPage.isTourDisplayed());
-        assertTrue(secondPage.hasResumeButton());
-
```

Comment:
Isn't this removing a test? Namely that we verify that if we go to another page that had a step in the tour, we have the resume button available and the tour can be resumed from that other page.